src/EventStore.Core/TransactionLog/Scavenging/DbAccess/ChunkWriterForExecutor.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using EventStore.Core.TransactionLog.Chunks;
+using EventStore.Core.TransactionLog.Chunks.TFChunk;
+using EventStore.Core.TransactionLog.LogRecords;
+
+namespace EventStore.Core.TransactionLog.Scavenging {
+	public class ChunkWriterForExecutor : IChunkWriterForExecutor<string, LogRecord> {
+		const int BatchLength = 2000;
+		private readonly ChunkManagerForExecutor _manager;
+		private readonly TFChunk _outputChunk;
+		private readonly List<List<PosMap>> _posMapss;
+		private int _lastFlushedPage = -1;
+		private long _fileSize = 0;
+
+		public ChunkWriterForExecutor(
+			ChunkManagerForExecutor manager,
+			TFChunkDbConfig dbConfig,
+			IChunkReaderForExecutor<string, LogRecord> sourceChunk) {
+
+			_manager = manager;
+
+			_posMapss = new List<List<PosMap>>(capacity: BatchLength) {
```

Comment:
is it possible to please add a comment about why a list of lists is needed here, i'm presuming it is to prevent the lists from going to the large object heap?

Reply:
yep that and growing the list becomes increasingly expensive because it has to copy out the whole list each time. i'll add a comment

Reply:
dpme

Reply:
thanks!